### PR TITLE
feat: Migrate most logic to useCodemirror

### PR DESCRIPTION
**Pull request by geoffgscott**
Moves most logic to the Codemirror hook for better encapsulation. 

Two way data binding is not currently working. Appears to be an issue where setting the spec does not update the visible codemirror content.

### CODE REVIEW

**Review by marclave — CHANGES_REQUESTED** [review]
awesome work! doing some testing and first round I found these 

- [ ] api client URL is missing
image
- [ ] API Client Response is missing 
image
- [ ] API Client Request Body doesnt change when switching endpoints 
- [ ] Show Schema doesnt work 
image
- [ ] operation Show Schema doesnt work 
- [ ] operation Response doesnt work 
image


will do a deeper dive after these meetings ✨

File: packages/use-codemirror/src/components/CodeMirror/CodeMirror.vue [hunk 1]
```diff
@@ -1,44 +1,12 @@
...
 import { useCodeMirror } from '../../hooks'
 import type { CodeMirrorLanguage } from '../../types'
-import { variables } from './extensions/variables'
 
 const props = withDefaults(
   defineProps<{
-    name?: string
-    extensions?: Extension[]
-    content?: string
+    content: string | undefined
```
Comment (hanspagel) [inline packages/use-codemirror/src/components/CodeMirror/CodeMirror.vue:9]: I don’t know what the difference is between those two. Can you enlighten me?

**Comment by hanspagel** [comment]
What an effort! Awesome work!

File: packages/use-codemirror/src/components/CodeMirror/CodeMirror.vue (hunk 1 above)
Comment (geoffgscott) [reply packages/use-codemirror/src/components/CodeMirror/CodeMirror.vue:9]: The first is an optional property. The second is a required property but the variable provided can be undefined. 

The advantage of the second is a user can't forget to assign a value to it.

**Review by marclave — APPROVED** [review]
awesome work! @geoffgscott 

can add a changeset and we can merge it ✨

**Pull request merged**